### PR TITLE
instancetype: Resolve TODO within hotplug func test

### PR DESCRIPTION
**Pull request by lyarwood**
### What this PR does

This PR addresses a simple TODO and starts using the WithInstancetype libvmi option within the instance type hot plug tests. The deprecated WithRunning is also replaced by WithRunStrategy.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

**Review by 0xFelix — COMMENTED** [review]
/approve

Thanks!

File: tests/hotplug/instancetype.go [hunk 1]
```diff
@@ -124,12 +121,7 @@ var _ = Describe("[sig-compute][Serial]Instance Type and Preference Hotplug", de
 		maxInstancetype, err = virtClient.VirtualMachineInstancetype(vmi.Namespace).Create(context.Background(), maxInstancetype, metav1.CreateOptions{})
 		Expect(err).ToNot(HaveOccurred())
 
-		// TODO - Move this into libvmi as a VMOption for the above NewVirtualMachine call
-		vm.Spec.Instancetype = &virtv1.InstancetypeMatcher{
-			Name: originalInstancetype.Name,
-			Kind: instancetypeapi.SingularResourceName,
-		}
-
+		vm = libvmi.NewVirtualMachine(vmi, libvmi.WithRunStrategy(virtv1.RunStrategyAlways), libvmi.WithInstancetype(originalInstancetype.Name))
```
Comment (fossedihelm) [inline tests/hotplug/instancetype.go:124]: nit: I think that with this change we can remove the global `vm` var. WDYT?

Comment (lyarwood) [reply tests/hotplug/instancetype.go]: ACK done, also removed some previously missed cruft that can be dropped now that we are using `WithInstancetype`.

**Comment by fossedihelm** [comment]
/lgtm
Thank you @lyarwood!

**Pull request merged**